src/disk_interface.cc

```diff
@@ -100,28 +101,34 @@ bool StatAllFilesInDir(const string& dir, map<string, TimeStamp>* stamps,
       static_cast<FINDEX_INFO_LEVELS>(1);
   FINDEX_INFO_LEVELS level =
       can_use_basic_info ? kFindExInfoBasic : FindExInfoStandard;
-  WIN32_FIND_DATAA ffd;
-  HANDLE find_handle = FindFirstFileExA((dir + "\\*").c_str(), level, &ffd,
-                                        FindExSearchNameMatch, NULL, 0);
+
+  std::string dir_string;
+  dir_string.reserve(dir.len_ + 3); // Make room for \* and null termination
```

Comment:
Are you certain that you need to reserve space for the nul terminator?

My understanding of the string api was that this was reserved for you already.